apps/mark-scan/pat-device-input/src/pin.rs

```diff
@@ -10,33 +10,54 @@ use vx_logging::{log, EventId};
 const EXPORT_PIN_FILEPATH: &str = "/sys/class/gpio/export";
 const UNEXPORT_PIN_FILEPATH: &str = "/sys/class/gpio/unexport";
 
+/// Pin provides an interface to read GPIO pins that support the sysfs protocol.
+///
+/// # Example
+///
+/// ```
+/// const PIN_NUMBER = 478;
+/// const MY_PIN: Pin = Pin::new(PIN_NUMBER);
+/// MY_PIN.set_up()?;
+/// let is_active = MY_PIN.is_active();
+/// do_something_with_value(is_active);
+/// ```
 pub struct Pin {
     address: u16,
 }
 
 impl Pin {
+    /// Creates a new instance of Pin at the given address (pin number).
     pub const fn new(address: u16) -> Self {
         Self { address }
     }
 
+    // Exports the pin to be globally accessible from userspace.
     fn export(&self) -> io::Result<()> {
         fs::write(EXPORT_PIN_FILEPATH, self.to_string())
     }
 
+    // Unexports the pin, removing access from userspace.
     fn unexport(&self) -> io::Result<()> {
         fs::write(UNEXPORT_PIN_FILEPATH, self.to_string())
     }
 
+    // Sets the pin direction to "in" so its value is readable. Must be called after
+    // `export`
     fn set_direction_in(&self) -> io::Result<()> {
         let filepath = format!("/sys/class/gpio/gpio{self}/direction");
         fs::write(filepath, b"in")
     }
 
+    /// Removes access to the pin from userspace.
     pub fn tear_down(&self) {
         self.unexport()
             .expect("Unexpected failure to tear down pin");
     }
 
+    /// Makes the pin accessible from userspace. If the pin is already set up due to
+    /// a previous `set_up` call, or from other callers of the sysfs interface, `set_up`
+    /// will tear down the pin and attempt one more time to set up. This may cause
+    /// interruption to other processes attempting to read the pin value.
```

Comment:
This brings up the question of whether we should just continue when we see the pin is already exported. IMO since we expect this daemon to be the only exporter it should never rely on the unexpected state of the pin already being exported. ie. it should always be exercising the path to export the pin.